### PR TITLE
[fix]:[CDS-68930]:Added validation for required fields

### PR DESCRIPTION
**Pull request by Rakshitharness**
## Describe your changes
Validation of required fields before creating delegate task for GcrArtifactResource and GARArtifactResource
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53064)

### CODE REVIEW

**Comment by Rakshitharness** [comment]
trigger messagecheck

**Comment by Rakshitharness** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java [hunk 1]
```diff
@@ -65,9 +65,10 @@ public class GARArtifactResource {
   @ApiOperation(
       value = "Gets google artifact registry build details", nickname = "getBuildDetailsForGoogleArtifactRegistry")
   public ResponseDTO<GARResponseDTO>
-  getBuildDetails(@QueryParam("connectorRef") String GCPConnectorIdentifier, @QueryParam("region") String region,
-      @QueryParam("repositoryName") String repositoryName, @QueryParam("project") String project,
-      @QueryParam("package") String pkg, @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
+  getBuildDetails(@QueryParam("connectorRef") String GCPConnectorIdentifier,
+      @NotNull @QueryParam("region") String region, @NotNull @QueryParam("repositoryName") String repositoryName,
+      @NotNull @QueryParam("project") String project, @NotNull @QueryParam("package") String pkg,
+      @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
```
Comment (Abhishek-CDC) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gar/GARArtifactResource.java:71]: We should not make changes to resource file. Adding @NotNull annotation may impact.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/googleartifactregistry/service/GARResourceServiceImpl.java [hunk 2]
```diff
@@ -105,6 +105,9 @@ public GARResourceServiceImpl(@Named(DEFAULT_CONNECTOR_SERVICE) ConnectorService
   public GARResponseDTO getBuildDetails(IdentifierRef googleArtifactRegistryRef, String region, String repositoryName,
       String project, String pkg, String version, String versionRegex, String orgIdentifier, String projectIdentifier) {
     GcpConnectorDTO connector = getConnector(googleArtifactRegistryRef);
+    ArtifactUtils.validateIfAllValuesAssigned(MutablePair.of(NGArtifactConstants.REGION, region),
+        MutablePair.of(NGArtifactConstants.REPOSITORY_NAME, repositoryName),
+        MutablePair.of(NGArtifactConstants.PROJECT, project), MutablePair.of(NGArtifactConstants.PACKAGE, pkg));
     BaseNGAccess baseNGAccess =
```
Comment (Abhishek-CDC) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/googleartifactregistry/service/GARResourceServiceImpl.java:111]: Please add unit tests

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/gcr/service/GcrResourceServiceImpl.java [hunk 3]
```diff
@@ -90,6 +90,8 @@ public GcrResourceServiceImpl(@Named(DEFAULT_CONNECTOR_SERVICE) ConnectorService
   @Override
   public GcrResponseDTO getBuildDetails(IdentifierRef gcrConnectorRef, String imagePath, String registryHostname,
       String orgIdentifier, String projectIdentifier) {
+    ArtifactUtils.validateIfAllValuesAssigned(MutablePair.of(NGArtifactConstants.IMAGE_PATH, imagePath),
+        MutablePair.of(NGArtifactConstants.REGISTRY_HOST_NAME, registryHostname));
```
Comment (Abhishek-CDC) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/gcr/service/GcrResourceServiceImpl.java:94]: lets create separate PR for GCR

**Comment by Rakshitharness** [comment]
trigger uts

**Comment by Rakshitharness** [comment]
trigger uts

**Review by naman-goenka — DISMISSED** [review]

**Comment by Rakshitharness** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

File: 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/gar/service/GarResourceServiceImplTest.java [hunk 4]
```diff
@@ -267,6 +269,89 @@ public void testFailureExecution() {
...
   }
 
+  @Test
+  @Owner(developers = RAKSHIT_AGARWAL)
+  @Category(UnitTests.class)
+  public void testGetBuildDetails_Region_Null() {
+    assertThatThrownBy(()
+                           -> garResourceService.getBuildDetails(CONNECTOR_REF, null, repositoryName, project, pkg,
+                               version, version_regex, ORG_IDENTIFIER, PROJECT_IDENTIFIER))
+        .isInstanceOf(InvalidRequestException.class)
```
Comment (Abhishek-CDC) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/gar/service/GarResourceServiceImplTest.java:279]: please add for runtimeInput cases also

**Review by Abhishek-CDC — APPROVED** [review]

**Comment by Rakshitharness** [comment]
trigger uts

**Comment by Rakshitharness** [comment]
trigger review

**Comment by Rakshitharness** [comment]
trigger uts

**Review by Abhishek-CDC — APPROVED** [review]

**Comment by Rakshitharness** [comment]
trigger review

**Review by naman-goenka — APPROVED** [review]

**Comment by Rakshitharness** [comment]
trigger review

**Pull request merged**